### PR TITLE
MEP-12: Rack Spreading

### PR DESCRIPTION
**Pull request by Gerrit91**

### CODE REVIEW

**Review by majst01 — APPROVED** [review]

File: docs/src/development/proposals/MEP12/partitioning.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+_backgroundColor: #1f1f1f
+_backgroundImage:
+_footer: ""
+-->
+![bg contain](partitioning-2.svg)
+
+---
+<style>section { font-size: 30px; }</style>
+
+# Eigenschaften Multi-Partition-Layout
```
Comment (mwennrich) [inline docs/src/development/proposals/MEP12/partitioning.md:38]: mix of german and english

File: docs/src/development/proposals/MEP12/partitioning.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+_backgroundColor: #1f1f1f
+_backgroundImage:
+_footer: ""
+-->
+![bg contain](partitioning-3.svg)
+
+---
+<style>section { font-size: 30px; }</style>
+
+# Eigenschaften Single-Partition-Layout
```
Comment (mwennrich) [inline docs/src/development/proposals/MEP12/partitioning.md:65]: same as above

File: docs/src/development/proposals/MEP12/partitioning.md (hunk 1 above)
Comment (Gerrit91) [reply docs/src/development/proposals/MEP12/partitioning.md]: Done. 🙈

File: docs/src/development/proposals/MEP12/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Spreading a group of machines across racks can enhance availability for scenarios like a rack loosing power or a chassis meltdown.
+
+So, instead of just randomly deciding the placement of a machine candidate, we want to propose the introduction of a placement strategy for the machine allocation.
+
+## Placement Strategies
+
+The following placement strategies should be implemented:
+
+- `project`: Machines in the project are spread across all available racks evenly (best effort), the user can optionally pass tags which will be considered for spreading the machines as well (this will for example allow spreading by a cluster id tag inside the same project)
+- `random`: Machines are randomly placed inside the racks (current implementation)
```
Comment (thheinel) [inline docs/src/development/proposals/MEP12/README.md:14]: - Why do you want to keep `"random"` if the default should be `"project"` (best effort) anyway?
- Why would a customer want to choose `"random"`?

Comment (Gerrit91) [reply docs/src/development/proposals/MEP12/README.md:14]: You are right. Probably the algorithm for `random` is simpler and faster, but I cannot come up with a direct advantage why a user should choose `random`. If nobody else finds an advantage of keeping the old strategy, I think we can remove it entirely.

Comment (thheinel) [reply docs/src/development/proposals/MEP12/README.md:14]: i think if a user needs new machines they will use the default settings no matter if it takes longer or not.

Comment (majst01) [reply docs/src/development/proposals/MEP12/README.md:14]: Yes lets get rid of the `random` strategy, there is no point keeping it, except we can see that machine allocation will take twice as long as before, which is very unlikely the case.

**Review by majst01 — APPROVED** [review]

**Pull request merged**